framework/src/timesteppers/FileTimeSequenceStepper.C

```diff
@@ -0,0 +1,43 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "FileTimeSequenceStepper.h"
+#include "MooseUtils.h"
+#include "libmesh/serial_mesh.h"
+#include "libmesh/exodusII_io.h"
+
+template<>
+InputParameters validParams<FileTimeSequenceStepper>()
+{
+  InputParameters params = validParams<TimeSequenceStepperBase>();
+  params.addRequiredParam<MeshFileName>("mesh", "The name of the mesh file to extract the time sequence from (must be an exodusII file).");
+  params.addClassDescription("Solves the Transient problem at a sequence of time points taken from a specified exodus file.");
+  return params;
+}
+
+FileTimeSequenceStepper::FileTimeSequenceStepper(const InputParameters & parameters) :
+    TimeSequenceStepperBase(parameters),
+    _mesh_file(getParam<MeshFileName>("mesh"))
+{
+  // Check that the required file exists
+  MooseUtils::checkFileReadable(_mesh_file);
+
+  // dummy mesh
+  SerialMesh mesh(_communicator);
+
+  // Read the Exodus file
+  ExodusII_IO exodusII_io(mesh);
+  exodusII_io.read(_mesh_file);
```

Comment:
Just be curious, is the entire mesh file read here? If so, it will be unnecessary.